### DISCUSSION TITLE
QZ connection to tacx trainer app

### DISCUSSION THREAD

**Discussion by oel59** [cagnulein/qdomyos-zwift#2327] [Q&A]
Hi,

I am an avid user of qz and use it to ride my domyos 900  with Zwift, Mywoosh, Rouvy
I was wondering if I could connect through qz to a tacx training app.
Tried it with my qz zwift setup but Tacx could not find my equipment.
does anyone have any experience with qz and tacx training g app?

thanks for your help.

leo.

**Comment by cagnulein** [comment]
Nope but I can do it if you want. It requires some days of work
Please join patreon if you didn't already
Thanks
https://www.patreon.com/cagnulein

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 10 mag 2024 alle 16:21 oel59 ***@***.***> ha
scritto:

**Reply by oel59** [reply]
Thank you Roberto for your quick reaction.

I have joined patreon in the meantime. I have sent you a message from the patreon platform.

Thanks and have a nice day,

Leo.

**Comment by cagnulein** [comment]
thanks i will let you know when i will be able to test it!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 11 mag 2024 alle 15:55 oel59 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
@oel59 one question: do you have ant+ on your android? If so there is an easy way to connect qz via ant+ to tacx app

**Reply by oel59** [reply]
Hi, I have ant+ on the tablet where I run tacx, but not sure of the samsung phone where I run qz. 
 Will check and let you know.

**Reply by oel59** [reply]
Yes, I have the ant+ services on the phone as well.

**Comment by cagnulein** [comment]
That's perfect! I'm near to a solution so! I will let you know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 13 mag 2024 alle 00:45 oel59 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
@oel59 yesterday I tried the ant+ connection from the tacx trainer app on android and I saw that it can't find any of my real ant+ sensors. Do you have the same experience? if so i have to implement this using bluetooth
Let me know

**Reply by oel59** [reply]
Hi Roberto,

I have the same experience.

**Reply by cagnulein** [reply]
ok what a crap of software :) ok let's go back to the original plan on bluetooth. Thanks for the confirmation

**Comment by cagnulein** [comment]
branch started https://github.com/cagnulein/qdomyos-zwift/pull/2336

**Reply by oel59** [reply]
Ciao Roberto,

Have you been able to test the connection with Tacx via BT?

Thanks,

Leo.

**Comment by cagnulein** [comment]
I started to but I'm sick right now. I just saw that simply setting the
tacx service is not enough. I will test some other ways in the next weeks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 26 mag 2024 alle 14:54 oel59 ***@***.***> ha
scritto:

**Comment by oel59** [comment]
Hi Roberto,

Get well first.
Tacx can wait.

Take care

Leo

**Comment by cagnulein** [comment]
this morning I tried all the way but the TACX app can't see at all the device. I'm writing a ticket to them, let's see how it goes. I will keep you updated

**Reply by cagnulein** [reply]
Garmin was useless about this. I'm trying other ways @oel59

**Comment by bogdanni** [comment]
Tacx trainer app seems to support only Tacx trainers. A way in for qz may be to masquerade as one of those trainers...

**Comment by cagnulein** [comment]
I already tried but it seems they are looking for something that I didn't
understood yet. If you find something useful let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 6 set 2024 alle 19:00 Bogdan Nicula ***@***.***>
ha scritto:

**Comment by bogdanni** [comment]
Not a wizard like you, but if you can get it to work, you can count on my Patreon support.

**Comment by cagnulein** [comment]
Thanks!


Il giorno ven 6 set 2024 alle 19:07 Bogdan Nicula ***@***.***>
ha scritto:

**Comment by supernrm** [comment]
@bogdanni 
> Tacx trainer app seems to support only Tacx trainers.

Yes, that is true with one small exception: somehow, it is possible to connect the Wahoo Kickr Core, but ONLY on the Windows version and ONLY via ANT. 

Yea, I know, crazy :D

**Reply by oel59** [reply]
Maybe QZ can pretend to be a Wahoo Kickr Core? 
Don't know if it would be so simple though.
It's really a pity that we cannot use the tacx training app from garmin.

**Reply by cagnulein** [reply]
I already investigated that way via bluetooth and the tacx doesn't accept it. I guess the only way would be through ANT+, but as @supernrm said it's available only on windows. You can check already if it works using QZ on android and using a ANT+ dongle enabling the ant cadence setting in QZ. I didn't test this yet

**Reply by supernrm** [reply]
I have a problem with resistance; it's not as smooth as on Zwift.  I set up the cassette  and the cog I'm using, but for example, when going downhill, I run out of gears even though I have 52/10 set. Hmm... Maybe I need setup more values in gear table.

**I confirm:**

-  Zwift Click changes my gears, I feel the resistance
**- I connected via QZ to Tacx on Windows!!!**
- HRM is separate

What's strange is that: power matches, RPM matches, but speed is much lower than in QZ, e.g., QZ shows 30 km/h, while Tacx shows 10 km/h?!

But yea, it's working! But please, don't tell tacx developers :D It's funny that they blocked every other trainers on every platform but not wahoo on windows :D

**Comment by cagnulein** [comment]
For speed don't worry about this, I mean the important is the wattage and
cadence

But I don't understand : what is the difference between zwift?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 6 mar 2025 alle 17:42 supernrm ***@***.***> ha
scritto:

**Reply by oel59** [reply]
I did a quick test:

- QZ on android sending data over ANT+
- Tacx app on my windows 11 laptop with an ant+ dongle and ant+ driver downloaded and installed

 I can confirm that cadence and power sensors of QZ can connect to the Tacx app.

I have a "simple" domyos 900 home trainer connected via bluetooth to QZ and I always use that setup with Zwift.
I noticed that the resistance settings are not changed on my bike, but this might need some more testing.

I did the same test with Tacx on my Android tablet, but then there is no connection. So, it only seems to work under windows.

A good step forward. Thank you supernrm for your input.

**Reply by cagnulein** [reply]
yes i didn't implement resistance over ANT+ but I can add in the next versions somehow

**Reply by oel59** [reply]
Thanks Roberto, that would be nice.

**Reply by cagnulein** [reply]
i'm doing it here https://github.com/cagnulein/qdomyos-zwift/pull/2330

**Reply by SagalSagal** [reply]
“Does QZ fully support Tacx Neo 2T on iOS?”

**Reply by cagnulein** [reply]
@SagalSagal Do you mean tacx trainer support or tacx app support? Tacx trainer yes it's supported on the ios version
